waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/core/federation/service/PopulateStatusFederationService.java

```diff
@@ -13,23 +13,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.hotels.bdp.waggledance.rest.service;
+package com.hotels.bdp.waggledance.core.federation.service;
 
 import java.util.ArrayList;
 import java.util.List;
 
+import org.springframework.stereotype.Service;
+
 import com.hotels.bdp.waggledance.api.federation.service.FederationService;
 import com.hotels.bdp.waggledance.api.federation.service.FederationStatusService;
 import com.hotels.bdp.waggledance.api.model.AbstractMetaStore;
 import com.hotels.bdp.waggledance.api.model.MetaStoreStatus;
+import com.hotels.bdp.waggledance.mapping.service.impl.NotifyingFederationService;
 
-public class DelegatingFederationService implements FederationService {
+@Service
+public class PopulateStatusFederationService implements FederationService {
 
-  private final FederationService federationService;
+  private final NotifyingFederationService federationService;
   private final FederationStatusService federationStatusService;
 
-  public DelegatingFederationService(
-      FederationService federationService,
+  public PopulateStatusFederationService(
+      NotifyingFederationService federationService,
```

Comment:
Would it be better to use the interface here to decouple these two classes?

```java
public PopulateStatusFederationService(
      FederationService federationService,
      FederationStatusService federationStatusService)

You may not even have to configure the bean because the only beans that can be injected here are this class itself and the `NotifyingFederationService`, so Spring might be able to inject for you.

Reply:
Will confirm if this works and Spring doesn't add `PopulateStatusFederationService` instead of `NotifyingFederationService` after I add some tests 😆

Reply:
Actually, IntelliJ gives an autowire error when I make the change from `NotifyingFederationService` to `FederationService`:

>      Could not autowire. There is more than one bean of 'FederationService' type.
>      Beans: notifyingFederationService    (NotifyingFederationService.java)
>             populateStatusFederationService    (PopulateStatusFederationService.java)

Reply:
you need to add @Qualifier("notifyingFederationService") if you want to use the interface

Reply:
Thanks, that worked :)